static/js/components/CommentForms.js

```diff
@@ -227,9 +229,30 @@ const mapDispatchToProps = (dispatch: any, ownProps: CommentFormProps) => {
         })
     }),
     patchComment: comment =>
-      dispatch(actions.comments.patch(comment.id, comment)).then(() => {
-        cancelReply(dispatch, formKey)()
-      }),
+      dispatch(actions.comments.patch(comment.id, comment))
+        .then(() => {
+          cancelReply(dispatch, formKey)()
+        })
+        .catch(err => {
+          dispatch(clearCommentError())
+          if (err.errorStatusCode === 410) {
+            // Comment was deleted
+            dispatch(
+              setBannerMessage(
+                "This comment has been deleted and cannot be edited"
+              )
+            )
+          } else {
+            // Unknown errors
+            dispatch(
+              setBannerMessage(
+                `Something went wrong editing your comment. Please try again or contact us at ${
+                  SETTINGS.support_email
+                }`
+              )
+            )
+          }
+        }),
```

Comment:
two questions about this function:

- I think it might be a little easier to read if it were written with the `async / await` sugar, up to you
- if the comment creation fails, should we also do `cancelReply` in that case, as well as in the success case? as it you see the banner message that your comment failed, but the form stays open. might prompt the user to spam submitting the comment, even if it won't ever work. what do you think?

Reply:
I rewrote the patch and post functions to use async/await.  I decided to use cancelReply if the error is `410` (comment deleted) only, because another error might be a temporary issue and the user might want to try again, or at least copy their reply text before it disappears and try again later.

Reply:
cool, just tried that out and it works well